### PR TITLE
feat: implement Firebase authentication middleware

### PR DESCRIPTION
**Pull request by fweilun**
- Replace auth middleware stub with real Firebase ID token verification
- Define TokenVerifier interface for testability; *auth.Client satisfies it
- Inject verified Firebase UID into request context via Auth(verifier) middleware
- Expose UserIDFromContext helper for downstream handlers
- Wire TokenVerifier through ServerDeps → NewRouter → middleware.Auth()
- Initialize Firebase auth client in main.go from FIREBASE_CREDENTIALS_JSON (nil verifier = dev mode passthrough with warning)
- Update all protected endpoints to read user_id from context instead of request body/query params: order, notification, calendar, location handlers
- Add unit tests with fakeVerifier covering valid token, missing header, invalid token, nil verifier, and context extraction cases

### CODE REVIEW

File: cmd/ark-api/main.go [hunk 1]
```diff
@@ -65,6 +69,23 @@ func main() {
 	calendarStore := calendar.NewStore(dbPool)
 	calendarSvc := calendar.NewService(calendarStore, orderSvc)
 
+	// Initialize Firebase auth client for token verification.
+	// If FIREBASE_CREDENTIALS_JSON is not set, auth middleware is disabled (dev mode).
+	var tokenVerifier middleware.TokenVerifier
+	if creds := cfg.Notification.FirebaseCredentialsJSON; creds != "" {
+		fbApp, err := firebase.NewApp(ctx, nil, option.WithCredentialsJSON([]byte(creds)))
```
[the remark this answers is not in the document]
Comment (fweilun) [reply cmd/ark-api/main.go]: The same Firebase project is used for both notifications and authentication.

File: internal/http/middleware/auth.go [hunk 2]
```diff
@@ -1,12 +1,57 @@
...
+			c.Next()
+			return
+		}
+		authHeader := c.GetHeader("Authorization")
+		if !strings.HasPrefix(authHeader, "Bearer ") {
+			c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": "missing or invalid Authorization header"})
+			return
+		}
+		idToken := strings.TrimPrefix(authHeader, "Bearer ")
+		token, err := verifier.VerifyIDToken(c.Request.Context(), idToken)
```
[the remark this answers is not in the document]
Comment (fweilun) [reply internal/http/middleware/auth.go]: No need to do this for now.
